packages/offix-datastore/src/DataStore.ts

```diff
@@ -35,30 +35,36 @@ export interface CustomEngines {
  * and GraphQL replication engine.
  */
 export class DataStore {
-  private models: Model<unknown>[];
   private storage: LocalStorage;
+  private replicator?: IReplicator;
 
   private config: DataStoreConfig;
-  private engines: CustomEngines | undefined;
 
   constructor(config: DataStoreConfig, engines?: CustomEngines) {
     this.config = config;
-    this.models = [];
-    this.engines = engines;
 
     if (engines && engines.storeAdapter) {
       this.storage = new LocalStorage(engines.storeAdapter);
     } else {
       const indexedDB = new IndexedDBStorageAdapter();
       this.storage = new LocalStorage(indexedDB);
     }
+
+    if (this.config.replicationConfig) {
+      if (engines?.replicator) {
+        this.replicator = engines.replicator;
+      } else {
+        this.replicator = new GraphQLCRUDReplicator(this.config.replicationConfig);
+      }
+    } else {
+      logger("Replication configuration was not provided. Replication will be disabled");
+    }
   }
 
   public createModel<T>(schema: DataSyncJsonSchema<T>) {
     const modelSchema = new ModelSchema(schema);
     const model = new Model<T>(modelSchema, this.storage);
-    // TODO replace any
-    this.models.push(model as any);
+    this.replicator?.startModelReplication(model, this.storage);
```

Comment:
So if we do this then we simply do not need replication init method. 
We can pass replication info to create model function..